provisioning/baremetal_pod.go

```diff
@@ -651,7 +716,7 @@ func CheckExistingMetal3Deployment(client appsclientv1.DeploymentsGetter, target
 func EnsureMetal3Deployment(info *ProvisioningInfo) (updated bool, err error) {
 	// Create metal3 deployment object based on current baremetal configuration
 	// It will be created with the cboOwnedAnnotation
-	metal3Deployment := newMetal3Deployment(info.Namespace, info.Images, &info.ProvConfig.Spec, info.PodLabelSelector)
+	metal3Deployment := newMetal3Deployment(info.Namespace, info.Images, &info.ProvConfig.Spec, info.PodLabelSelector, info.Proxy)
```

Comment:
Since we are are passing in several items already stored in info and now we are adding the proxy config to this list, maybe we should just pass ProvisioningInfo as a parameter. This PR can continue to merge without this change but would be nice to see this change in this or a subsequent PR.

Reply:
That makes sense, but would prefer to change it separately to keep this PR from getting too big.